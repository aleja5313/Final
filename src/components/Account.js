import React, { useEffect, useState } from "react";
import { useDispatch } from "react-redux";
import { db } from "../firebase/firebaseConfig";
import {
  collection,
  getDocs,
  deleteDoc,
  doc,
  setDoc,
} from "firebase/firestore";
import { Button } from "react-bootstrap";
import { async } from "@firebase/util";

const Account = () => {
  const [users, setUsers] = useState([]);
  const [flag, setFlag] = useState(true);
  const [edad, setEdad] = useState([]);

  useEffect(() => {
    const getUsers = async () => {
      const usersCollectionRef = collection(db, "users");
      const data = await getDocs(usersCollectionRef);
      const users1 = data.docs.map((doc) => ({ ...doc.data(), id: doc.id }));

      setUsers(users1);
    };
    getUsers();
  }, [flag]);

  const handleDelete = async (id) => {
    const docRef = doc(db, "users", id);
    await deleteDoc(docRef);
    setFlag(!flag);
  };

  const addAge = async (id, index, nombre, correo) => {
    const docRef = doc(db, "users", id);
    const payload = {
      nombre,
      correo,
      acceso: true,
      edad: edad[index],
    };
    await setDoc(docRef, payload);

    setFlag(!flag);
  };
  const handleChange = (event) => {
    const { value, id } = event.target;
    const edad1 = [...edad];
    edad1[id] = value;
    setEdad(edad1);
  };

  useEffect(() => {
    let edadInicial = [];
    for (let index = 0; index < 100; index++) {
      edadInicial = [...edadInicial, 0];
    }
    setEdad(edadInicial);
  }, []);
  return (
    <div>
      <table className="table table-dark">
        <thead>
          <tr>
            <th scope="col">#</th>
            <th scope="col">Nombre</th>
            <th scope="col">Correo</th>
            <th scope="col">Edad</th>
            <th scope="col">Borrar</th>
          </tr>
        </thead>
        <tbody>
          {users.map((user, index) => {
            return (
              <tr key={user.id}>
                <th scope="row">{index + 1}</th>
                <td>{user.nombre}</td>
                <td>{user.correo}</td>
                <td>
                  {user.edad ? (
                    user.edad
                  ) : (
                    <div>
                      <input
                        id={index}
                        type="text"
                        value={edad[index]}
                        onChange={handleChange}
                      />
                      <Button
                        onClick={() =>
                          addAge(user.id, index, user.nombre, user.correo)
                        }
                      >
                        <svg
                          xmlns="http://www.w3.org/2000/svg"
                          width="16"
                          height="16"
                          fill="currentColor"
                          className="bi bi-pen"
                          viewBox="0 0 16 16"
                        >
                          <path d="m13.498.795.149-.149a1.207 1.207 0 1 1 1.707 1.708l-.149.148a1.5 1.5 0 0 1-.059 2.059L4.854 14.854a.5.5 0 0 1-.233.131l-4 1a.5.5 0 0 1-.606-.606l1-4a.5.5 0 0 1 .131-.232l9.642-9.642a.5.5 0 0 0-.642.056L6.854 4.854a.5.5 0 1 1-.708-.708L9.44.854A1.5 1.5 0 0 1 11.5.796a1.5 1.5 0 0 1 1.998-.001zm-.644.766a.5.5 0 0 0-.707 0L1.95 11.756l-.764 3.057 3.057-.764L14.44 3.854a.5.5 0 0 0 0-.708l-1.585-1.585z" />
                        </svg>
                      </Button>
                    </div>
                  )}
                </td>
                <td>
                  <Button id={index} onClick={(e) => handleDelete(index)}>
                    Eliminar
                  </Button>
                </td>
              </tr>
            );
          })}
        </tbody>
      </table>
    </div>
  );
};

export default Account;
