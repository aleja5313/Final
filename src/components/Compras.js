import React, { useEffect, useState } from "react";
import { Container } from "../styles/style.jsx";
import swal from "sweetalert";

export const Compras = () => {
  const [ingredientes, setIngredientes] = useState([]);

  const [compras, setCompras] = useState({});

  const [cantidades, setCantidades] = useState([]);

  const [valorTotal, setValorTotal] = useState(0);

  const [cantidadItems, setCantidadItems] = useState(0);

  const mostartAlerta = () => {
    swal("Compra con éxito");
  };

  useEffect(() => {
    const f = async () => {
      const data = await getData();
      setIngredientes(data);
      const i = data.map((ingrediente) => {
        return 1;
      });
      setCantidades(i);
    };
    f();
  }, []);

  useEffect(() => {
    let contador = 0;
    let contadorItems = 0;
    Object.keys(compras).forEach((compra) => {
      console.log(compras[compra]);
      const { id, precio } = compras[compra];
      contador += cantidades[id] * precio;
      contadorItems += cantidades[id];
    });
    contador = (Math.round(contador * 100) / 100).toFixed(2);
    setValorTotal(contador);
    setCantidadItems(contadorItems);
  }, [compras, cantidades]);

  const select1 = (event) => {
    const { checked, id, name } = event.target;
    const seleccionados = { ...compras };
    if (checked) {
      seleccionados[name] = {
        id,
        precio: ingredientes[id].price,
      };
    } else {
      delete seleccionados[name];
    }
    setCompras(seleccionados);
  };

  const actualizarCantidades = (event) => {
    const { id } = event.target;

    const a = [...cantidades];

    a[parseInt(id)] += 1;

    setCantidades(a);
  };
  return (
    <Container id="c">
      <h3>INGREDIENTES</h3>
      <h1>Risotto de setas vegano</h1>
      <input type="checkbox" name="" id="cex" />
      Seleccionar todo / Deseleccionar todo
      {ingredientes.map((ingrediente, index) => {
        return (
          <div id="t" key={index}>
            <br />
            <input
              className="check"
              onChange={select1}
              indice={index}
              type="checkbox"
              id={index}
              name={ingrediente.product}
            />
            <label htmlFor="tentacles"> </label>
            <input
              value={cantidades[index]?.toString() || "1"}
              onChange={actualizarCantidades}
              type="number"
              id={index}
              name="tentacles"
              min="1"
              max="100"
            ></input>
            {ingrediente.product}
            <br />
            {ingrediente.brand}
            <br />
            {ingrediente.quantity}
            <br />
            {ingrediente.price}
          </div>
        );
      })}
      <br />
      <h4 id="resultadof">Items {cantidadItems}</h4>
      <h4>Subtotal {valorTotal} $</h4>
      <h4>Gastos de Envío 7,00 $</h4>
      <h2>Total {(Math.round(valorTotal * 7 * 100) / 100).toFixed(2)} $</h2>
      <button onClick={() => mostartAlerta()}>
        Comprar ingredientes:{" "}
        {(Math.round(valorTotal * 7 * 100) / 100).toFixed(2)} $
      </button>
    </Container>
  );
};

export function getData() {
  let url = "https://recipe-rissoto.vercel.app/recipe";

  let settings = {
    method: "GET",
  };

  return new Promise((resolve) => {
    fetch(url, settings)
      .then((res) => res.json())
      .then((json) => {
        resolve(json.ingredients);
      });
  });
}
