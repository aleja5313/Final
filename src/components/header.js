import '../styles/style.jsx';
import {NavBar, Navbar,NavBarUl, Container,} from '../styles/style.jsx'

function Header() {
  return (

    <div>
  <NavBar>
    <img src="https://i.ibb.co/4dWXnRn/images-q-tbn-ANd9-Gc-Rr-Mm-f-YIj-Yhm-UMAr64-GAc-Fp-Ine521w2p-L-6-A-usqp-CAU.png" alt="" width="70px" ></img>
    <NavBarUl >
          <a href='./Compras'>Home</a>
              <a href='./Login'>Sign in</a>
              <a href='./Account'>Account</a>
    </NavBarUl>
  </NavBar>

  </div>

  )
}

export default Header