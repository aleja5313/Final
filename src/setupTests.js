import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {createSerializer} from 'enzyme-to-json';
import fetchMock from "jest-fetch-mock";

fetchMock.enableMocks();
Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));
