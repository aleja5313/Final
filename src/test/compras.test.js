import React from "react";
import { shallow, mount } from "enzyme";
import { Compras, getData } from "../components/Compras";
import "../setupTests";
import { act } from "react-dom/test-utils";

describe("Compras", () => {
  beforeEach(() => {
    try {
      fetch.mockResponseOnce(JSON.stringify(mock));
    } catch (error) {
      console.log(error);
    }
  });

    it("Deberia renderizar correctamente", () => {
      const component = shallow(<Compras />);

      expect(component).toMatchSnapshot();
    });

  it("Deberia consultar la API correctamente", async () => {
    let data;
    try {
      data = await getData();
    } catch (error) {
      data = {};
      console.log(error);
    }

    expect(data).toEqual(mock.ingredients);
  });

  it("Valor inicial deberia ser 0", async () => {
    await act(async () => {
      const component1 = mount(<Compras />);

      expect(component1.find("#c").find("#resultadof").text()).toBe("Items 0");
    });
  });
});

const mock = {
  ingredients: [
    {
      product: "producto de prueba",
      brand: "marca de prueba",
      items: 1,
      quantity: "600 gr.",
      price: 2.95,
    },
  ],
};
