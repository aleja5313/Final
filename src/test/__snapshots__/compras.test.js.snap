// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compras Deberia renderizar correctamente 1`] = `
<styled.div
  id="c"
>
  <h3>
    INGREDIENTES
  </h3>
  <h1>
    Risotto de setas vegano
  </h1>
  <input
    id="cex"
    name=""
    type="checkbox"
  />
  Seleccionar todo / Deseleccionar todo
  <br />
  <h4
    id="resultadof"
  >
    Items 
    0
  </h4>
  <h4>
    Subtotal 
    0
     $
  </h4>
  <h4>
    Gastos de Env√≠o 7,00 $
  </h4>
  <h2>
    Total 
    0.00
     $
  </h2>
  <button
    onClick={[Function]}
  >
    Comprar ingredientes:
     
    0.00
     $
  </button>
</styled.div>
`;
